src/crc32.rs

```diff
@@ -111,6 +121,14 @@ impl Crc32 {
 
     /// Consume the HAL peripheral, returning the PAC peripheral
     pub fn free(self) -> CRC {
+        unsafe {
+            // NOTE(unsafe) this reference will only be used for atomic writes with no side effects.
+            let rcc_raw = unsafe { &(*RCC::ptr()) };
```

Comment:
This is already inside a unsafe block, so the second unsafe is unnecessary and emits a warning.

Reply:
Oops! will correct.